scripts/test_deploy.sh

```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
+###########
+#### SCRIPT: DEPLOYMENT from local repository into SWARM for testing
+#### PARAMS: see -h
+###########
+#USAGE: deployment based on local repository
+
+ask_continue () {
+    while true; do
+        
+        read -p "$1 (y or n): " input        
+        case $input in
+            [yY]*)
+                echo '---Continuing!'
+                break
+                ;;
+            [nN]*)
+                echo '***So do THAT, exiting!'
+                exit 1
+                ;;
+             *)
+                echo '***Invalid input' >&2
+        esac
+    done
+}
+
+echo_dec ()
+{
+    echo "---------------------------------------"
+    echo "--> $1"
+    echo "---------------------------------------"
+}
+
+
+er () {
+    echo_dec "${1}"
+    eval "${1}"
+    ret_val=$?
+    echo "....................."
+    echo ">> DONE: ${1} with status: ${ret_val}"
+    return $ret_val
+}
+
+check_create_network () {
+    
+    docker network inspect ${network_name} > /dev/null 2>&1
+    network_status=$?
+    if [ ${network_status} -eq 1  ]; then
+      er "docker network create -d overlay ${network_name}"
+      echo_dec "Network ${network_name} created"
+    else
+       echo_dec "Network ${network_name} already exists"
+    fi
+}
+
+check_stack_exists ()
+{
+    docker stack ps ${stack_name} > /dev/null 2>&1
+    stack_status=$?
+    return ${stack_status}
+}
+
+actualize_repo ()
+{
+
+    valid_ans=false
+
+    while [ "$valid_ans" = false ]; do
+        
+        echo "Do you want to update the local repository to origin/dev first before creating a docker image? (y/n)"   
+        read ans
+
+        if [ "$ans" = "y" ] || [ "$ans" = "Y" ]; then
+            echo "Update repository..."
+            git reset --hard origin/dev
```

Comment:
Nemělo by tady být ještě git pull?

Reply:
Potřeba to není, ale pro jistotu přidám git fetch origin